scripts/discover-repos.sh

```diff
@@ -0,0 +1,456 @@
+#!/bin/bash
+#
+# discover-repos.sh - Auto-discover Go repositories in konflux-ci organization
+#
+# This script:
+# 1. Finds all Go repositories in konflux-ci org
+# 2. Detects repository ownership (CODEOWNERS or top contributors)
+# 3. Suggests exclude patterns based on repository structure
+# 4. Generates configuration files for new repositories
+# 5. Updates CODEOWNERS file with ownership mappings
+#
+# Usage:
+#   ./scripts/discover-repos.sh                      # Dry run - shows what would be added
+#   ./scripts/discover-repos.sh --apply              # Actually creates files
+#   ./scripts/discover-repos.sh --create-pr-per-repo # Creates individual PR per repository
+#
+
+set -e
+
+SCRIPT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+ROOT_DIR="$(dirname "$SCRIPT_DIR")"
+REPOS_DIR="$ROOT_DIR/repos"
+CODEOWNERS_FILE="$ROOT_DIR/CODEOWNERS"
+OUTPUT_DIR="$ROOT_DIR/discovered-repos"
+
+# Modes
+DRY_RUN=true
+CREATE_PR_PER_REPO=false
+
+# Parse arguments
+while [[ $# -gt 0 ]]; do
+  case $1 in
+    --apply)
+      DRY_RUN=false
+      shift
+      ;;
+    --create-pr-per-repo)
+      DRY_RUN=false
+      CREATE_PR_PER_REPO=true
+      shift
+      ;;
+    *)
+      echo "Unknown option: $1"
+      echo "Usage: $0 [--apply] [--create-pr-per-repo]"
+      exit 1
+      ;;
+  esac
+done
+
+echo "========================================="
+echo "Konflux-CI Repository Auto-Discovery"
+echo "========================================="
+echo
+echo "Mode: $([ "$DRY_RUN" = true ] && echo "DRY RUN (use --apply to create files)" || echo "APPLY - will create files")"
+echo
+
+# Check prerequisites
+if ! command -v jq &> /dev/null; then
+  echo "❌ Error: jq is required but not installed"
+  exit 1
+fi
+
+if ! command -v gh &> /dev/null && [ "$CREATE_PR_PER_REPO" = true ]; then
+  echo "❌ Error: gh CLI is required for --create-pr-per-repo but not installed"
```

Comment:
This error message isn't correct. The `gh` CLI isn't only used when this flag is set.

Please make sure to review everything these AI agents spit out for accuracy.